implement StorageImpl with fixed-size array and support for null keys

Mwxyz2022 opened the pull request:
- Used Object[][] array to store key-value pairs (max 10 items)
- Implemented put(), get(), and size() methods
- Added logic to update value if key already exists
- get() method supports null keys and uses @SuppressWarnings to handle cast
- Added helper method getItemIndex() to find key index in storage

Mwxyz2022 committed:
implement StorageImpl with fixed-size array and support for null keys

- Used Object[][] array to store key-value pairs (max 10 items)
- Implemented put(), get(), and size() methods
- Added logic to update value if key already exists
- get() method supports null keys and uses @SuppressWarnings to handle cast
- Added helper method getItemIndex() to find key index in storage

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,56 @@
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEM = 10;
+    private final Object[][] storage;
```
Issue: The storage array is declared as `Object[][] storage`, but in the constructor, only the outer array is initialized (`new Object[MAX_ITEM][]`). According to the checklist, each element should be an array of size 2 (for key and value). You should initialize the storage as `new Object[MAX_ITEM][2]` to avoid `NullPointerException` when accessing `storage[i][1]`.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your work! There is a critical issue with how the storage array is initialized in your `StorageImpl` class. Currently, only the outer array is initialized, but the inner arrays (which should each hold a key and value) are not. This can cause a `NullPointerException` when you try to access or assign values to `storage[i][1]`. To fix this, you should initialize the storage as `new Object[10][2]`, so that each element is a two-element array for key and value. Please address this issue to meet the core requirements and avoid runtime errors. You're almost there—keep it up! 🚀

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Mwxyz2022 committed:
improve storage logic and null key handling in StorageImpl

- Initialized storage with fixed inner size: new Object[10][2]
- Replaced dynamic size() call with countFillItems to track real count
- Limited getItemIndex() loop to filled slots only
- Fixed bug where null keys could cause incorrect overwriting or access

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 Your implementation meets all the core requirements and constraints of the task. The logic for storing, retrieving, and updating key-value pairs is correct, and your code is clean and functional. There are no critical issues or missing functionalities. Keep up the good work and continue following best practices!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

Mwxyz2022 committed:
- Fixed spacing

Elena-Bruyako commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,51 @@
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEM = 10;
+    private final Object[][] storage;
+    private int countFillItems = 0;
```
```suggestion
    private int size = 0;
```

Elena-Bruyako commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,51 @@
+
+    private int getItemIndex(K key) {
+        for (int i = 0; i < countFillItems; i++) {
+            if ((key == null && storage[i][0] == null)
+                    || (key != null && key.equals(storage[i][0]))) {
```
```suggestion
            if (key == storage[i][0]
                    || key != null && key.equals(storage[i][0])) {
```

Mwxyz2022 committed:
- rename countFillItems to size for consistency

liliia-ponomarenko commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,51 @@
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEM = 10;
+    private final Object[][] storage;
+    private int size = 0;
```
```suggestion
    private int size;
```
default value

liliia-ponomarenko commented on src/main/java/core/basesyntax/impl/StorageImpl.java:
```diff
@@ -3,17 +3,51 @@
+    private int size = 0;
+
+    public StorageImpl() {
+        this.storage = new Object[MAX_ITEM][2];
```
two-dimensional array is not really a clear solution. Better have separate arrays or create an object that represents the product. but in general it is work. so you can try refactor it if you want.

liliia-ponomarenko reviewed (approved):
good job!
